src/components/forms/ProductFiltersForm/ProductFiltersForm.tsx

```diff
@@ -30,40 +34,44 @@ const ProductFiltersForm = () => {
     resolver: yupResolver(productFiltersSchema),
     defaultValues: {
       priceRange: filters.priceRange,
-      ratingRange: filters.ratingRange
+      ratingRange: filters.ratingRange,
+      categories: filters.categories
     }
   });
   const [searchParams, setSearchParams] = useSearchParams();
 
   useEffect(() => {
     const params: any = Object.fromEntries(searchParams.entries());
-    const {minPrice, maxPrice, minRating, maxRating} = params;
+    const {minPrice, maxPrice, minRating, maxRating, categories} = params;
     if (minPrice && maxPrice) setValue('priceRange', [+minPrice, +maxPrice]);
     if (minRating && maxRating) setValue('ratingRange', [+minRating, +maxRating]);
+    if (categories) setValue('categories', decodeParamsArray(categories));
     const formValues = getValues();
     dispatch(updateFilters(formValues));
   }, []);
 
   const updateSearchParams = (filters: Filters): void => {
-    const {priceRange, ratingRange} = filters;
-    const params: any = {
-      minPrice: priceRange[0],
-      maxPrice: priceRange[1],
-      minRating: ratingRange[0],
-      maxRating: ratingRange[1]
-    };
+    const {priceRange, ratingRange, categories} = filters;
+    let params: any = {};
+    if (priceRange[0] !== undefined) params.minPrice = priceRange[0];
+    if (priceRange[1] !== undefined) params.maxPrice = priceRange[1];
+    if (ratingRange[0] !== undefined) params.minRating = ratingRange[0];
```

Comment:
What is the minimum value for price and rating '0' or '1' you are working with? I suggest finding the more beautiful solution instead of `if (priceRange[0] !== undefined)`. There are too much `if` checking.